integration/generic-tests/kexec_upl_test.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2024 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build !race
+// +build !race
+
+package integration
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/hugelgupf/vmtest/qemu"
+	"github.com/hugelgupf/vmtest/scriptvm"
+	"github.com/hugelgupf/vmtest/testtmp"
+	"github.com/u-root/mkuimage/uimage"
+)
+
+func TestKexecUpl(t *testing.T) {
+	qemu.SkipIfNotArch(t, qemu.ArchAMD64, qemu.ArchArm64)
+
+	script := `
+		kexec /upl
+	`
+
+	initrd := filepath.Join(testtmp.TempDir(t), "initramfs.cpio")
+	vm := scriptvm.Start(t, "vm", script,
+		scriptvm.WithUimage(
+			// Build kexec as a binary command to get accurate GOCOVERDIR
+			// integration coverage data (busybox rewrites command code).
+			uimage.WithCoveredCommands("github.com/u-root/u-root/cmds/core/kexec"),
+			uimage.WithFiles(fmt.Sprintf("%s:upl", os.Getenv("VMTEST_UPL"))),
```

Comment:
If you have a docker image with just the VMTEST_UPL data, this is a good place to insert that so the test can run automatically: https://github.com/u-root/u-root/blob/main/.vmtest.yaml

(See the vmtest README for how to use)

Reply:
Let's fix all other issues and ensure the whole process works as expected. After that, we can go back to the integration test and fit it into vmtests. (maybe in another pr ?)

Reply:
I would much prefer that the feature + test get merged in one PR. That way nothing can be break it between now and later.

To build a Docker image with VMTEST_UPL -- copy this template for a generic edk2 build: https://github.com/u-root/u-root/blob/main/.circleci/images/uefipayload-amd64/Dockerfile

And set up a Github action by adding it to the list here: https://github.com/u-root/u-root/blob/main/.github/workflows/test-images.yml

Then Github Actions will build a new image for you, and you can use it in this PR and automatically have it pulled in by the vmtest framework like so: https://github.com/u-root/u-root/blob/main/.vmtest.yaml#L21